crates/consensus/src/transaction/deposit.rs

```diff
@@ -241,6 +243,28 @@ impl Decodable for TxDeposit {
     }
 }
 
+impl SignableTransaction<Signature> for TxDeposit {
+    fn set_chain_id(&mut self, _chain_id: ChainId) {}
+
+    fn payload_len_for_signature(&self) -> usize {
+        self.encoded_len(false)
+    }
+
+    fn encode_for_signing(&self, out: &mut dyn alloy_rlp::BufMut) {
+        self.encode_inner(out, false);
+    }
+
+    fn into_signed(self, signature: Signature) -> alloy_consensus::Signed<Self> {
+        let payload_length = self.encoded_len(false) + signature.rlp_vrs_len();
+        let mut out = Vec::with_capacity(payload_length);
+        self.encode_inner(&mut out, false);
+        signature.write_rlp_vrs(&mut out);
+        let hash = keccak256(out);
```

Comment:
this will produce invalid hash. deposit transaction rlp encoding does not include a signature